components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/model/OpenAPIRequest.java

```diff
@@ -90,12 +91,16 @@ public OpenAPIRequest(MessageContext messageContext) {
                 .stream()
                 .collect(Collectors.toMap(Map.Entry::getKey, entry -> Collections.singleton(entry.getValue())));
         //Set transport headers
-        String contentTypeHeader = "content-type";
         for (Map.Entry<String, Collection<String>> header : headerMap.entrySet()) {
             String headerKey = header.getKey();
-            String value =  header.getValue().iterator().next();
-            headerKey = headerKey.equalsIgnoreCase(contentTypeHeader) ?
-                    "Content-Type" : headerKey.toLowerCase(Locale.ROOT);
+            String value = header.getValue().iterator().next();
+            if (headerKey.equalsIgnoreCase(Headers.CONTENT_TYPE)) {
```

Comment:
Shall we switch the equality check to avoid NPEs?
```suggestion
            if (Headers.CONTENT_TYPE.equalsIgnoreCase(headerKey)) {